http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java

```diff
@@ -511,6 +516,90 @@ public void testPutMethodWithFileBodyGenerator()
         assertTrue(testFile.delete());
     }
 
+    @Test
+    public void testPutMethodWithFileStreamingBodyGenerator()
+            throws Exception
+    {
+        byte[] contents = "hello world, I'm streaming at you".getBytes(UTF_8);
+        File testFile = File.createTempFile("test", null);
+        Files.write(testFile.toPath(), contents);
+
+        URI uri = baseURI.resolve("/road/to/nowhere");
+        Request request = preparePut()
+                .setUri(uri)
+                .addHeader(CONTENT_TYPE, "x-test")
+                .setBodyGenerator(streamingBodyGenerator(new FileInputStream(testFile)))
+                .build();
+
+        int statusCode = executeRequest(request, createStatusResponseHandler()).getStatusCode();
+        assertEquals(statusCode, 200);
+        assertEquals(servlet.getRequestMethod(), "PUT");
+        assertEquals(servlet.getRequestUri(), uri);
+        assertEquals(servlet.getRequestHeaders(CONTENT_TYPE), ImmutableList.of("x-test"));
+        assertEquals(servlet.getRequestBytes(), contents);
+
+        assertTrue(testFile.delete());
+    }
+
+    @Test
+    public void testPutMethodWithStreamingBodyGenerator()
+            throws Exception
+    {
+        String responseBody = IntStream.range(0, 10000).mapToObj(ignore -> "hello").collect(Collectors.joining());
```

Comment:
This is so small that I'm concerned we're not really testing the streaming behavior.

Reply:
I'll change to `100_000` hellos. That should be enough.